File: Java/O(N)/ON_128.java
// Source: https://www.geeksforgeeks.org/find-repetitive-element-1-n-1/

// Java program to find the only repeating
// element in an array where elements are
// from 1 to N-1.
class ON_128 {

	static int findRepeating(int arr[], int N)
	{

		// res is going to store value of
		// 1 ^ 2 ^ 3 .. ^ (N-1) ^ arr[0] ^
		// arr[1] ^ .... arr[n-1]
		int res = 0;
		for (int i = 0; i < N - 1; i++)
			res = res ^ (i + 1) ^ arr[i];
		res = res ^ arr[N - 1];

		return res;
	}

	// Driver code
	public static void main(String[] args)
	{
		int arr[] = { 9, 8, 2, 6, 1, 8, 5, 3, 4, 7 };
		int N = arr.length;
	
		// Function call
		System.out.println(findRepeating(arr, N));
	}
}

// This code is contributed by
// Smitha Dinesh Semwal.

File: Python/O(N)/on_128.py
# Source: https://www.geeksforgeeks.org/maximum-equilibrium-sum-in-an-array/

# Python3 program to find
# maximum equilibrium sum.

# Function to find maximum
# equilibrium sum.
def findMaxSum(arr, n):

	# Array to store prefix sum.
	preSum = [0 for i in range(n)]

	# Array to store suffix sum.
	suffSum = [0 for i in range(n)]

	# Variable to store maximum sum.
	ans = -10000000

	# Calculate prefix sum.
	preSum[0] = arr[0]
	
	for i in range(1, n):
	
		preSum[i] = preSum[i - 1] + arr[i]

	# Calculate suffix sum and compare
	# it with prefix sum. Update ans
	# accordingly.
	suffSum[n - 1] = arr[n - 1]
	if (preSum[n - 1] == suffSum[n - 1]):
		ans = max(ans, preSum[n - 1])
	
	for i in range(n - 2, -1, -1):
		suffSum[i] = suffSum[i + 1] + arr[i]
		if (suffSum[i] == preSum[i]):
			ans = max(ans, preSum[i])
	
	return ans

# Driver Code
if __name__=='__main__':

	arr = [-2, 5, 3, 1,2, 6, -4, 2]
	n = len(arr)
	print(findMaxSum(arr, n))
	
# This code i contributed by pratham76
